addons/account/wizard/account_move_send.py

```diff
@@ -384,7 +403,10 @@ def _prepare_invoice_pdf_report(self, invoices_data):
         :param invoice_data:    The collected data for the invoice so far.
         """
         ids = [inv.id for inv in invoices_data]
-        content, _report_type = self.env['ir.actions.report']._pre_render_qweb_pdf('account.account_invoices', res_ids=ids)
+
+        pdf_report_id = next(value['pdf_report_id'] for value in invoices_data.values() if 'pdf_report_id' in value)
```

Comment:
You need to split the data per `pdf_report_id`. Suppose:
- send invoice A and B using the send & print wizard asynchronously using pdf_report_id Y
- send invoice C and D using the send & print wizard asynchronously using pdf_report_id Z
- run the CRON: you currently generate all invoices with Y

Reply:
I am not sure I get what u mean, but for the case of having 2 different `pdf_report_id` I could not reproduce it.
However, I handled this case in a separate commit. Once u review it and confirm it is what u meant, I can merge the 2 commits together :)